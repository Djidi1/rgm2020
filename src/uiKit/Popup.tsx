import React, { FC } from 'react';
import styled from 'styled-components';

type MainProps = {
  show: boolean,
  children: JSX.Element[] | JSX.Element | string,
  setShowPopup: (arg: boolean) => void,
};

interface PopupProps {
  hide: boolean
}

const PopupCloseButton = styled.span`
  color: #fff;
  position: absolute;
  top: 0.4em;
  right: 0.5em;
  font-size: 0.7rem;
  cursor: pointer;
`;

const PopUpMenu = styled.div<PopupProps>`
  position: absolute;
  right: 1em;
  top: 1em;
  padding: 1em 0;
  z-index: 100;
  background: #232323;
  display: ${({ hide }) => (hide ? 'none' : 'flex')};
  flex-direction: column;
  border-radius: 4px;
  & button {
    background: transparent;
    border: none;
    color: #fff;
    cursor: pointer;
    width: 100px;
    padding: 0.5em 1em;
    text-align: left;
    :hover {
      background: #f65261;
    }
  }
`;

const Dialog: FC<MainProps> = ({
  show, children, setShowPopup,
}) => (
  <PopUpMenu hide={!show}>
    <PopupCloseButton onClick={() => setShowPopup(false)}>X</PopupCloseButton>
    {children}
  </PopUpMenu>
);

export default Dialog;
