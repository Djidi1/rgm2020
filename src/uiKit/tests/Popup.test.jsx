import React from 'react';
import renderer from 'react-test-renderer';
import Popup from '../Popup';

describe('UI-Kit Popup', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('is expected to render successfully', () => {
    const wrapper = renderer
      .create(<Popup />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
