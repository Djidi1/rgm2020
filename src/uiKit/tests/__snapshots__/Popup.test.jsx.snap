// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI-Kit Popup is expected to render successfully 1`] = `
<div
  className="Popup__PopUpMenu-sc-1xszb4m-1 iMSOXG"
>
  <span
    className="Popup__PopupCloseButton-sc-1xszb4m-0 hLuUyq"
  >
    X
  </span>
</div>
`;
